src/lib/net/classes/remote.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * !TODO: Add server/client differentation, yeah I know this should have made from the very first start, but lol,
+ * that will take a bit of my brain, but i already got something planned out for this.
+ */
+
+import { BaseEvent } from "../types";
+
+export declare enum EAction {
+	COND = "condition",
+	DO = "do",
+}
+
+function isAFunction(func: (() => boolean) | boolean): func is Callback {
+	return typeOf(func) === "function";
+}
+
+export class RemoteClass<T extends unknown[]> implements BaseEvent<T> {
+	protected listeners: Array<[Callback | boolean, EAction]> = new Array();
```

Comment:
Nope, these conditions are not evaluated, you're being passed an evaluable condition. Generally not a great way of defining this

Reply:
```ts
type CondStep = [ Callback, EAction.COND ]
type FuncStep = [ Callback, EAction.DO ]

// ...

protected listeners: Array<CondStep | FuncStep> = [];
```
Oh yeah also, no `new Array()` constructor, just use `[]`

Reply:
I like to keep with the `new T` part as I use it for maps, seems like a good way to keep it up. Anyway, what do you mean with conditions not being evaluated? What is the purpose of the condition then?